src/python/shorah_dec.py

```diff
@@ -25,6 +25,11 @@
     file of corrected reads
 '''
 
+from __future__ import print_function
+from __future__ import division
+from builtins import dict
+from future.utils import listitems
+from builtins import range
```

Comment:
@ozagordi that's pretty much what susie did

1. future (without the leading and trailing underscores) is the package you also referenced, it monkey patches a number of python modules at runtime
2. \_\_future__ (with the leading and trailing underscores) is a core python module specified in PEP 236 (https://www.python.org/dev/peps/pep-0236/)

Susie ported most of the code to proper py3 idioms, with the future+builtins providing support for py2, so if you were to drop py2 support, you would only need minimal changes.